src/networkmanager.cpp

```diff
@@ -128,26 +131,98 @@ void NetworkManager::request_document(MSDocumentType document_type,
 
 void NetworkManager::connect_to_server(server_type p_server)
 {
-  server_socket->close();
-  server_socket->abort();
+  disconnect_from_server();
 
   qInfo().nospace().noquote() << "connecting to " << p_server.ip << ":"
                               << p_server.port;
 
-  server_socket->connectToHost(p_server.ip, p_server.port);
+  switch (p_server.socket_type) {
+  default:
+      p_server.socket_type = TCP;
```

Comment:
```suggestion
    p_server.socket_type = TCP;
    [[fallthrough]];
```
clang-tidy had one job